src/floats.jl

```diff
@@ -339,6 +339,9 @@ const BIGFLOAT = [BigFloat()]
         # this and the next branch are pretty slow for BigInt
         # but also extremely rare (i.e. floats w/ > 39 digits)
         x = v / exp10(V(-exp))
+    elseif exp == 23
+        # special-case due to https://github.com/JuliaLang/julia/issues/38509
+        x = v * V(1e23)
```

Comment:
Note that, because `1e23` is not exactly representable as a `Float64`, this will introduce errors for any floating-point type that is wider than `Float64`.

Reply:
Is there a better representation that would be more accurate? We currently don't support any wide floating point types than `Float64`, but it'd be nice to know anyway.

Reply:
You could use `exp10` for every type other than `Float16/32/64`…

(For example, I can imagine people wanting to read CSV files into decimal floats via DecFP.jl in order to preserve exactness.  Although decimal floats are slow enough that maybe you might as well use DelimitedFiles.)